### PR TITLE
instruction_durations overwrites backend durations, does not extend them

### PR DESCRIPTION
**Pull request by ajavadia**
like all other args in transpile(), supplying your own arg should overwrite the one that is read from backend.

plus a couple other enhancements to the InstructionDuration object, like a print option.

### CODE REVIEW

**Comment by itoko** [comment]
Thank you. Looks good to me. I leave a few minor comments.

File: qiskit/compiler/transpile.py [hunk 1]
```diff
@@ -631,14 +631,17 @@ def _parse_scheduling_method(scheduling_method, num_circuits):
...
-            durations = InstructionDurations(inst_durations, dt)
+    if inst_durations is None and backend:
+        try:
+            backend_durations = InstructionDurations.from_backend(backend)
+        except AttributeError:
+            backend_durations = InstructionDurations()
+        durations = backend_durations.update(inst_durations, dt)
+    else:
+        durations = InstructionDurations(inst_durations,
+                dt or getattr(inst_durations, 'dt', None))
```
Comment (itoko) [inline qiskit/compiler/transpile.py:644]: indent error here

File: qiskit/compiler/transpile.py [hunk 2]
```diff
@@ -631,14 +631,17 @@ def _parse_scheduling_method(scheduling_method, num_circuits):
...
-        if backend:
-            durations = InstructionDurations.from_backend(backend).update(inst_durations, dt)
-        else:
-            durations = InstructionDurations(inst_durations, dt)
+    if inst_durations is None and backend:
+        try:
+            backend_durations = InstructionDurations.from_backend(backend)
+        except AttributeError:
+            backend_durations = InstructionDurations()
+        durations = backend_durations.update(inst_durations, dt)
```
Comment (itoko) [inline qiskit/compiler/transpile.py:641]: I see this line is still necessary to update `dt`. `durations = backend_durations.update(None, dt)` might be clearer.

File: qiskit/compiler/transpile.py [hunk 3]
```diff
@@ -631,14 +631,17 @@ def _parse_scheduling_method(scheduling_method, num_circuits):
...
-def _parse_instruction_durations(backend, inst_durations, dt, scheduling_method, num_circuits):
+def _parse_instruction_durations(backend, inst_durations, dt, num_circuits):
     durations = None
-    if scheduling_method is not None:
-        from qiskit.transpiler.instruction_durations import InstructionDurations
-        if backend:
-            durations = InstructionDurations.from_backend(backend).update(inst_durations, dt)
-        else:
-            durations = InstructionDurations(inst_durations, dt)
+    if inst_durations is None and backend:
```
Comment (itoko) [inline qiskit/compiler/transpile.py:636]: Now `InstructionDurations` is always created if backend is specified even when it is not used (`scheduling_method=None`). Is it too tiny overhead to care?

Comment (ajavadia) [reply qiskit/compiler/transpile.py:636]: yeah i think it's tiny

**Review by itoko — APPROVED** [review]

**Pull request merged**